graphz/src/test/scala/coop/rchain/graphz/GraphzSpec.scala

```diff
@@ -0,0 +1,293 @@
+        _ <- g.close
+      } yield g
+
+      val graph = for {
+        g <- Graphz[Effect]("Process", DiGraph)
+        _ <- g.node("0")
+        _ <- g.subgraph(process1)
+        _ <- g.edge("0", "A")
+        _ <- g.subgraph(process2)
+        _ <- g.edge("0", "K")
+        _ <- g.node("1")
+        _ <- g.edge("M", "1")
+        _ <- g.edge("C", "1")
+        _ <- g.close
+      } yield g
+      graph.show shouldBe (
+        """digraph Process {
+          |  "0" []
+          |  subgraph cluster_p1 {
+          |    label = "process #1"
+          |    color=blue
+          |    "A" []
+          |    "B" []
+          |    "C" []
+          |    A -> B []
+          |    B -> C []
+          |  }
+          |  0 -> A []
+          |  subgraph cluster_p2 {
+          |    label = "process #2"
+          |    color=green
+          |    "K" []
+          |    "L" []
+          |    "M" []
+          |    K -> L []
+          |    L -> M []
+          |  }
+          |  0 -> K []
+          |  "1" []
+          |  M -> 1 []
+          |  C -> 1 []
+          |}""".stripMargin
+      )
+    }
+
+    // from https://github.com/xflr6/graphviz/blob/master/examples/process.py
+    it("Process example") {
+      (for {
+        graph <- Graphz[Effect]("G", Graph)
+        _     <- graph.edge("run", "intr")
+        _     <- graph.edge("intr", "runbl")
+        _     <- graph.edge("runbl", "run")
+        _     <- graph.edge("run", "kernel")
+        _     <- graph.edge("kernel", "zombie")
+        _     <- graph.edge("kernel", "sleep")
+        _     <- graph.edge("kernel", "runmem")
+        _     <- graph.edge("sleep", "swap")
+        _     <- graph.edge("swap", "runswap")
+        _     <- graph.edge("runswap", "new")
+        _     <- graph.edge("runswap", "runmem")
+        _     <- graph.edge("new", "runmem")
+        _     <- graph.edge("sleep", "runmem")
+        _     <- graph.close
+      } yield graph).show shouldBe (
+        """graph G {
+          |  run -- intr []
+          |  intr -- runbl []
+          |  runbl -- run []
+          |  run -- kernel []
+          |  kernel -- zombie []
+          |  kernel -- sleep []
+          |  kernel -- runmem []
+          |  sleep -- swap []
+          |  swap -- runswap []
+          |  runswap -- new []
+          |  runswap -- runmem []
+          |  new -- runmem []
+          |  sleep -- runmem []
+          |}""".stripMargin
+      )
+    }
+
+  }
+
+  implicit class GraphzOps(graph: Effect[Graphz[Effect]]) {
+    def show: String =
+      graph.runS(new StringBuffer).toString
+
+    import java.io.File
+    import java.io.PrintWriter
+
+    def view(): Unit = {
+      val sourcePath = "/Users/rabbit/temp.gv"
+      val outputPath = "/Users/rabbit/output.pdf"
+      new File(sourcePath).createNewFile()
+      val writer = new PrintWriter(sourcePath)
+      writer.println(show)
+      writer.flush()
+      writer.close
+      val dotCmd  = s"dot -Tpdf $sourcePath -o $outputPath"
```

Comment:
What if `$sourcePath` has spaces in it?

Reply:
this is just in tests for debugging purposes - this whole method will be removed